trainer/lib/trainer/test_parser.rb

```diff
@@ -197,12 +197,36 @@ def execute_cmd(cmd)
       return output
     end
 
+    # Hotfix: From Xcode 16 beta 3 'xcresulttool get --format json' has been deprecated;
+    #         '--legacy' flag required to keep on using the command
+    def generate_cmd_parse_xcresult(path)
+      xcresulttool_cmd = %W(
+        xcrun
+        xcresulttool
+        get
+        --format
+        json
+        --path
+        #{path}
+      )
+
+      # Version of xcresulttool bundled in Xcode 16 beta 3 was 23021
+      xcresulttool_version_string = `xcrun xcresulttool version`.split(',')[0]
```

Comment:
Could you note the expected output from `xcrun xcresulttool version` on the comment?

```ruby
# xcresulttool version 23021, format version 3.53 (current)
```

Reply:
I prefer to use regex and the named capture for code readability

```ruby
match = result.match(/xcresulttool version (?<version>\d+)/)
version = match[:version]&.to_f
```